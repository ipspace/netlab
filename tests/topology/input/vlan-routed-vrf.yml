addressing:
  unnumbered:
    ipv4: True
    ipv6: True

vrfs:
  red:
  blue:
  green:

vlans:
  red:
    vrf: red
  blue:
    vrf: blue
  green:
    vrf: green
    mode: route
    role: unnumbered

groups:
  hosts:
    members: [ h1, h2, h3 ]
    module: []
    device: frr

module: [ vlan, vrf ]

provider: clab

nodes:
  r1:
    device: frr
    vlans:
      red:
        mode: route
      blue:
        mode: route
  r2:
    device: frr
    vlans:
      red:
        mode: route
      blue:
        mode: route
      green:
        mode: route
  s1:
    device: eos
    vlans:
      green:
        mode: bridge
  s2:
    device: eos
  h1:                   # red VLAN
  h2:                   # blue VLAN
  h3:                   # green VLAN

links:
- s1:
    vlan.access: red
  h1:
- s2:
    vlan.access: blue
  h2:
- s2:
    vlan.access: green
    vlan.mode: bridge
  h3:
- s1:
  s2:
    vlan.mode: bridge
  vlan.trunk: [ red, green, blue ]
- s1:
  r1:
    vlan.mode: route
  vlan.trunk: [ red, green, blue ]
- s2:
    vlan.mode: bridge
  r2:
  vlan.trunk: [ red, green, blue ]
