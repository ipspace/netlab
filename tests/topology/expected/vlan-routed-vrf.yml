groups:
  hosts:
    device: frr
    members:
    - h1
    - h2
    - h3
    module: []
input:
- topology/input/vlan-routed-vrf.yml
- package:topology-defaults.yml
links:
- bridge: input_1
  interfaces:
  - ifindex: 1
    ifname: Ethernet1
    ipv4: 172.16.0.3/24
    node: s1
    vlan:
      access: red
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.0.5/24
    node: h1
  linkindex: 1
  node_count: 2
  prefix:
    ipv4: 172.16.0.0/24
  type: lan
  vrf: red
- bridge: input_2
  interfaces:
  - ifindex: 1
    ifname: Ethernet1
    ipv4: 172.16.1.4/24
    node: s2
    vlan:
      access: blue
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.1.6/24
    node: h2
  linkindex: 2
  node_count: 2
  prefix:
    ipv4: 172.16.1.0/24
  type: lan
  vrf: blue
- bridge: input_3
  interfaces:
  - ifindex: 2
    ifname: Ethernet2
    node: s2
    vlan:
      access: green
      mode: bridge
  - ifindex: 1
    ifname: eth1
    ipv4: true
    ipv6: true
    node: h3
  linkindex: 3
  node_count: 2
  prefix:
    ipv4: true
    ipv6: true
  type: lan
  vrf: green
- interfaces:
  - ifindex: 2
    ifname: Ethernet2
    node: s1
    vlan:
      trunk:
        blue: {}
        green: {}
        red: {}
  - ifindex: 3
    ifname: Ethernet3
    node: s2
    vlan:
      mode: bridge
      trunk:
        blue: {}
        green: {}
        red: {}
  linkindex: 4
  node_count: 2
  prefix: {}
  type: p2p
  vlan:
    trunk:
      blue: {}
      green: {}
      red: {}
- interfaces:
  - ifindex: 3
    ifname: Ethernet3
    node: s1
    vlan:
      trunk:
        blue: {}
        green: {}
        red: {}
  - ifindex: 1
    ifname: eth1
    node: r1
    vlan:
      mode: route
      trunk:
        blue: {}
        green: {}
        red: {}
  linkindex: 5
  node_count: 2
  prefix: {}
  type: p2p
  vlan:
    trunk:
      blue: {}
      green: {}
      red: {}
- interfaces:
  - ifindex: 4
    ifname: Ethernet4
    node: s2
    vlan:
      mode: bridge
      trunk:
        blue: {}
        green: {}
        red: {}
  - ifindex: 1
    ifname: eth1
    node: r2
    vlan:
      trunk:
        blue: {}
        green: {}
        red: {}
  linkindex: 6
  node_count: 2
  prefix: {}
  type: p2p
  vlan:
    trunk:
      blue: {}
      green: {}
      red: {}
module:
- vlan
- vrf
name: input
nodes:
  h1:
    af:
      ipv4: true
    box: frrouting/frr:v8.3.1
    clab:
      binds:
        clab_files/h1/daemons: /etc/frr/daemons
      config_templates:
        daemons: /etc/frr/daemons
      kind: linux
    device: frr
    hostname: clab-input-h1
    id: 5
    interfaces:
    - bridge: input_1
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.0.5/24
      linkindex: 1
      mtu: 1500
      name: h1 -> [r1,r2,s1,s2]
      neighbors:
      - ifname: eth1.1000
        ipv4: 172.16.0.1/24
        node: r1
      - ifname: eth1.1000
        ipv4: 172.16.0.2/24
        node: r2
      - ifname: Vlan1000
        ipv4: 172.16.0.3/24
        node: s1
      - ifname: Vlan1000
        node: s2
      type: lan
    loopback:
      ipv4: 10.0.0.5/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.105
      mac: 08-4F-A9-00-00-05
    module: []
    mtu: 1500
    name: h1
  h2:
    af:
      ipv4: true
    box: frrouting/frr:v8.3.1
    clab:
      binds:
        clab_files/h2/daemons: /etc/frr/daemons
      config_templates:
        daemons: /etc/frr/daemons
      kind: linux
    device: frr
    hostname: clab-input-h2
    id: 6
    interfaces:
    - bridge: input_2
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.1.6/24
      linkindex: 2
      mtu: 1500
      name: h2 -> [r1,r2,s2,s1]
      neighbors:
      - ifname: eth1.1001
        ipv4: 172.16.1.1/24
        node: r1
      - ifname: eth1.1001
        ipv4: 172.16.1.2/24
        node: r2
      - ifname: Vlan1001
        ipv4: 172.16.1.4/24
        node: s2
      - ifname: Vlan1001
        ipv4: 172.16.1.3/24
        node: s1
      type: lan
    loopback:
      ipv4: 10.0.0.6/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.106
      mac: 08-4F-A9-00-00-06
    module: []
    mtu: 1500
    name: h2
  h3:
    af:
      ipv4: true
      ipv6: true
    box: frrouting/frr:v8.3.1
    clab:
      binds:
        clab_files/h3/daemons: /etc/frr/daemons
      config_templates:
        daemons: /etc/frr/daemons
      kind: linux
    device: frr
    hostname: clab-input-h3
    id: 7
    interfaces:
    - bridge: input_3
      ifindex: 1
      ifname: eth1
      ipv4: true
      ipv6: true
      linkindex: 3
      mtu: 1500
      name: h3 -> [r1,r2,s2,s1]
      neighbors:
      - ifname: eth1.1002
        ipv4: true
        ipv6: true
        node: r1
      - ifname: eth1.1002
        ipv4: true
        ipv6: true
        node: r2
      - ifname: Vlan1002
        node: s2
      - ifname: Vlan1002
        node: s1
      type: lan
    loopback:
      ipv4: 10.0.0.7/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.107
      mac: 08-4F-A9-00-00-07
    module: []
    mtu: 1500
    name: h3
  r1:
    af:
      ipv4: true
      ipv6: true
      vpnv4: true
      vpnv6: true
    box: frrouting/frr:v8.3.1
    clab:
      binds:
        clab_files/r1/daemons: /etc/frr/daemons
      config_templates:
        daemons: /etc/frr/daemons
      kind: linux
    device: frr
    hostname: clab-input-r1
    id: 1
    interfaces:
    - ifindex: 1
      ifname: eth1
      linkindex: 5
      mtu: 1500
      name: r1 -> s1
      neighbors:
      - ifname: Ethernet3
        node: s1
      subif_index: 3
      type: p2p
      vlan:
        mode: route
    - bridge_group: 1
      ifindex: 2
      ifname: eth1.1001
      ipv4: 172.16.1.1/24
      name: r1 -> [r2,s2,s1,h2]
      neighbors:
      - ifname: eth1.1001
        ipv4: 172.16.1.2/24
        node: r2
      - ifname: Vlan1001
        ipv4: 172.16.1.4/24
        node: s2
      - ifname: Vlan1001
        ipv4: 172.16.1.3/24
        node: s1
      - ifname: eth1
        ipv4: 172.16.1.6/24
        node: h2
      parent_ifindex: 1
      parent_ifname: eth1
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1001
        mode: route
      vrf: blue
    - bridge_group: 2
      ifindex: 3
      ifname: eth1.1002
      ipv4: true
      ipv6: true
      name: r1 -> [r2,s2,s1,h3]
      neighbors:
      - ifname: eth1.1002
        ipv4: true
        ipv6: true
        node: r2
      - ifname: Vlan1002
        node: s2
      - ifname: Vlan1002
        node: s1
      - ifname: eth1
        ipv4: true
        ipv6: true
        node: h3
      parent_ifindex: 1
      parent_ifname: eth1
      pool: unnumbered
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1002
        mode: route
      vrf: green
    - bridge_group: 3
      ifindex: 4
      ifname: eth1.1000
      ipv4: 172.16.0.1/24
      name: r1 -> [r2,h1,s1,s2]
      neighbors:
      - ifname: eth1.1000
        ipv4: 172.16.0.2/24
        node: r2
      - ifname: eth1
        ipv4: 172.16.0.5/24
        node: h1
      - ifname: Vlan1000
        ipv4: 172.16.0.3/24
        node: s1
      - ifname: Vlan1000
        node: s2
      parent_ifindex: 1
      parent_ifname: eth1
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1000
        mode: route
      vrf: red
    loopback:
      ipv4: 10.0.0.1/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.101
      mac: 08-4F-A9-00-00-01
    module:
    - vlan
    - vrf
    mtu: 1500
    name: r1
    vlan:
      max_bridge_group: 3
    vlans:
      blue:
        bridge_group: 1
        id: 1001
        mode: route
        prefix:
          ipv4: 172.16.1.0/24
        vni: 101001
        vrf: blue
      green:
        bridge_group: 2
        id: 1002
        mode: route
        pool: unnumbered
        prefix:
          ipv4: true
          ipv6: true
        vni: 101002
        vrf: green
      red:
        bridge_group: 3
        id: 1000
        mode: route
        prefix:
          ipv4: 172.16.0.0/24
        vni: 101000
        vrf: red
    vrf:
      as: 65000
    vrfs:
      blue:
        af:
          ipv4: true
        export:
        - '65000:2'
        id: 2
        import:
        - '65000:2'
        rd: '65000:2'
        vrfidx: 101
      green:
        af:
          ipv4: true
          ipv6: true
        export:
        - '65000:3'
        id: 3
        import:
        - '65000:3'
        rd: '65000:3'
        vrfidx: 102
      red:
        af:
          ipv4: true
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        rd: '65000:1'
        vrfidx: 100
  r2:
    af:
      ipv4: true
      ipv6: true
      vpnv4: true
      vpnv6: true
    box: frrouting/frr:v8.3.1
    clab:
      binds:
        clab_files/r2/daemons: /etc/frr/daemons
      config_templates:
        daemons: /etc/frr/daemons
      kind: linux
    device: frr
    hostname: clab-input-r2
    id: 2
    interfaces:
    - ifindex: 1
      ifname: eth1
      linkindex: 6
      mtu: 1500
      name: r2 -> s2
      neighbors:
      - ifname: Ethernet4
        node: s2
      subif_index: 3
      type: p2p
    - bridge_group: 1
      ifindex: 2
      ifname: eth1.1001
      ipv4: 172.16.1.2/24
      name: r2 -> [r1,s2,s1,h2]
      neighbors:
      - ifname: eth1.1001
        ipv4: 172.16.1.1/24
        node: r1
      - ifname: Vlan1001
        ipv4: 172.16.1.4/24
        node: s2
      - ifname: Vlan1001
        ipv4: 172.16.1.3/24
        node: s1
      - ifname: eth1
        ipv4: 172.16.1.6/24
        node: h2
      parent_ifindex: 1
      parent_ifname: eth1
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1001
        mode: route
      vrf: blue
    - bridge_group: 2
      ifindex: 3
      ifname: eth1.1002
      ipv4: true
      ipv6: true
      name: r2 -> [r1,s2,s1,h3]
      neighbors:
      - ifname: eth1.1002
        ipv4: true
        ipv6: true
        node: r1
      - ifname: Vlan1002
        node: s2
      - ifname: Vlan1002
        node: s1
      - ifname: eth1
        ipv4: true
        ipv6: true
        node: h3
      parent_ifindex: 1
      parent_ifname: eth1
      pool: unnumbered
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1002
        mode: route
      vrf: green
    - bridge_group: 3
      ifindex: 4
      ifname: eth1.1000
      ipv4: 172.16.0.2/24
      name: r2 -> [r1,h1,s1,s2]
      neighbors:
      - ifname: eth1.1000
        ipv4: 172.16.0.1/24
        node: r1
      - ifname: eth1
        ipv4: 172.16.0.5/24
        node: h1
      - ifname: Vlan1000
        ipv4: 172.16.0.3/24
        node: s1
      - ifname: Vlan1000
        node: s2
      parent_ifindex: 1
      parent_ifname: eth1
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1000
        mode: route
      vrf: red
    loopback:
      ipv4: 10.0.0.2/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.102
      mac: 08-4F-A9-00-00-02
    module:
    - vlan
    - vrf
    mtu: 1500
    name: r2
    vlan:
      max_bridge_group: 3
    vlans:
      blue:
        bridge_group: 1
        id: 1001
        mode: route
        prefix:
          ipv4: 172.16.1.0/24
        vni: 101001
        vrf: blue
      green:
        bridge_group: 2
        id: 1002
        mode: route
        pool: unnumbered
        prefix:
          ipv4: true
          ipv6: true
        vni: 101002
        vrf: green
      red:
        bridge_group: 3
        id: 1000
        mode: route
        prefix:
          ipv4: 172.16.0.0/24
        vni: 101000
        vrf: red
    vrf:
      as: 65000
    vrfs:
      blue:
        af:
          ipv4: true
        export:
        - '65000:2'
        id: 2
        import:
        - '65000:2'
        rd: '65000:2'
        vrfidx: 101
      green:
        af:
          ipv4: true
          ipv6: true
        export:
        - '65000:3'
        id: 3
        import:
        - '65000:3'
        rd: '65000:3'
        vrfidx: 102
      red:
        af:
          ipv4: true
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        rd: '65000:1'
        vrfidx: 100
  s1:
    af:
      ipv4: true
      vpnv4: true
    box: ceos:4.26.4M
    clab:
      env:
        INTFTYPE: et
      kind: ceos
    device: eos
    hostname: clab-input-s1
    id: 3
    interfaces:
    - bridge: input_1
      clab:
        name: et1
      ifindex: 1
      ifname: Ethernet1
      linkindex: 1
      type: lan
      vlan:
        access: red
        access_id: 1000
    - clab:
        name: et2
      ifindex: 2
      ifname: Ethernet2
      linkindex: 4
      name: s1 -> s2
      neighbors:
      - ifname: Ethernet3
        node: s2
      type: p2p
      vlan:
        trunk:
          blue: {}
          green: {}
          red: {}
        trunk_id:
        - 1000
        - 1002
        - 1001
    - clab:
        name: et3
      ifindex: 3
      ifname: Ethernet3
      linkindex: 5
      name: s1 -> r1
      neighbors:
      - ifname: eth1
        node: r1
      type: p2p
      vlan:
        trunk:
          blue: {}
          green: {}
          red: {}
        trunk_id:
        - 1000
        - 1002
        - 1001
    - bridge_group: 1
      ifindex: 10
      ifname: Vlan1000
      ipv4: 172.16.0.3/24
      name: VLAN red (1000) -> [r1,r2,h1,s2]
      neighbors:
      - ifname: eth1.1000
        ipv4: 172.16.0.1/24
        node: r1
      - ifname: eth1.1000
        ipv4: 172.16.0.2/24
        node: r2
      - ifname: eth1
        ipv4: 172.16.0.5/24
        node: h1
      - ifname: Vlan1000
        node: s2
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
      vrf: red
    - bridge_group: 2
      ifindex: 11
      ifname: Vlan1001
      ipv4: 172.16.1.3/24
      name: VLAN blue (1001) -> [r1,r2,s2,h2]
      neighbors:
      - ifname: eth1.1001
        ipv4: 172.16.1.1/24
        node: r1
      - ifname: eth1.1001
        ipv4: 172.16.1.2/24
        node: r2
      - ifname: Vlan1001
        ipv4: 172.16.1.4/24
        node: s2
      - ifname: eth1
        ipv4: 172.16.1.6/24
        node: h2
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
      vrf: blue
    - bridge_group: 3
      ifindex: 12
      ifname: Vlan1002
      name: VLAN green (1002) -> [r1,r2,s2,h3]
      neighbors:
      - ifname: eth1.1002
        ipv4: true
        ipv6: true
        node: r1
      - ifname: eth1.1002
        ipv4: true
        ipv6: true
        node: r2
      - ifname: Vlan1002
        node: s2
      - ifname: eth1
        ipv4: true
        ipv6: true
        node: h3
      pool: unnumbered
      type: svi
      virtual_interface: true
      vlan:
        mode: bridge
      vrf: green
    loopback:
      ipv4: 10.0.0.3/32
    mgmt:
      ifname: Management0
      ipv4: 192.168.121.103
      mac: 08-4F-A9-00-00-03
    module:
    - vlan
    - vrf
    name: s1
    vlan:
      max_bridge_group: 3
    vlans:
      blue:
        bridge_group: 2
        id: 1001
        mode: irb
        neighbors:
        - ifname: eth1.1001
          ipv4: 172.16.1.1/24
          node: r1
        - ifname: eth1.1001
          ipv4: 172.16.1.2/24
          node: r2
        - ifname: Vlan1001
          ipv4: 172.16.1.4/24
          node: s2
        - ifname: Vlan1001
          ipv4: 172.16.1.3/24
          node: s1
        - ifname: eth1
          ipv4: 172.16.1.6/24
          node: h2
        prefix:
          ipv4: 172.16.1.0/24
        vni: 101001
        vrf: blue
      green:
        bridge_group: 3
        id: 1002
        mode: bridge
        pool: unnumbered
        prefix:
          ipv4: true
          ipv6: true
        vni: 101002
        vrf: green
      red:
        bridge_group: 1
        id: 1000
        mode: irb
        neighbors:
        - ifname: eth1.1000
          ipv4: 172.16.0.1/24
          node: r1
        - ifname: eth1.1000
          ipv4: 172.16.0.2/24
          node: r2
        - ifname: eth1
          ipv4: 172.16.0.5/24
          node: h1
        - ifname: Vlan1000
          ipv4: 172.16.0.3/24
          node: s1
        - ifname: Vlan1000
          node: s2
        prefix:
          ipv4: 172.16.0.0/24
        vni: 101000
        vrf: red
    vrf:
      as: 65000
    vrfs:
      blue:
        af:
          ipv4: true
        export:
        - '65000:2'
        id: 2
        import:
        - '65000:2'
        rd: '65000:2'
        vrfidx: 101
      green:
        export:
        - '65000:3'
        id: 3
        import:
        - '65000:3'
        rd: '65000:3'
        vrfidx: 102
      red:
        af:
          ipv4: true
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        rd: '65000:1'
        vrfidx: 100
  s2:
    af:
      ipv4: true
      vpnv4: true
    box: ceos:4.26.4M
    clab:
      env:
        INTFTYPE: et
      kind: ceos
    device: eos
    hostname: clab-input-s2
    id: 4
    interfaces:
    - bridge: input_2
      clab:
        name: et1
      ifindex: 1
      ifname: Ethernet1
      linkindex: 2
      type: lan
      vlan:
        access: blue
        access_id: 1001
    - bridge: input_3
      clab:
        name: et2
      ifindex: 2
      ifname: Ethernet2
      linkindex: 3
      type: lan
      vlan:
        access: green
        access_id: 1002
        mode: bridge
    - clab:
        name: et3
      ifindex: 3
      ifname: Ethernet3
      linkindex: 4
      name: s2 -> s1
      neighbors:
      - ifname: Ethernet2
        node: s1
      type: p2p
      vlan:
        mode: bridge
        trunk:
          blue: {}
          green: {}
          red: {}
        trunk_id:
        - 1000
        - 1002
        - 1001
    - clab:
        name: et4
      ifindex: 4
      ifname: Ethernet4
      linkindex: 6
      name: s2 -> r2
      neighbors:
      - ifname: eth1
        node: r2
      type: p2p
      vlan:
        mode: bridge
        trunk:
          blue: {}
          green: {}
          red: {}
        trunk_id:
        - 1000
        - 1002
        - 1001
    - bridge_group: 1
      ifindex: 11
      ifname: Vlan1001
      ipv4: 172.16.1.4/24
      name: VLAN blue (1001) -> [r1,r2,s1,h2]
      neighbors:
      - ifname: eth1.1001
        ipv4: 172.16.1.1/24
        node: r1
      - ifname: eth1.1001
        ipv4: 172.16.1.2/24
        node: r2
      - ifname: Vlan1001
        ipv4: 172.16.1.3/24
        node: s1
      - ifname: eth1
        ipv4: 172.16.1.6/24
        node: h2
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
      vrf: blue
    - bridge_group: 2
      ifindex: 12
      ifname: Vlan1002
      name: VLAN green (1002) -> [r1,r2,s1,h3]
      neighbors:
      - ifname: eth1.1002
        ipv4: true
        ipv6: true
        node: r1
      - ifname: eth1.1002
        ipv4: true
        ipv6: true
        node: r2
      - ifname: Vlan1002
        node: s1
      - ifname: eth1
        ipv4: true
        ipv6: true
        node: h3
      type: svi
      virtual_interface: true
      vlan:
        mode: route
      vrf: green
    - bridge_group: 3
      ifindex: 13
      ifname: Vlan1000
      name: VLAN red (1000) -> [r1,r2,h1,s1]
      neighbors:
      - ifname: eth1.1000
        ipv4: 172.16.0.1/24
        node: r1
      - ifname: eth1.1000
        ipv4: 172.16.0.2/24
        node: r2
      - ifname: eth1
        ipv4: 172.16.0.5/24
        node: h1
      - ifname: Vlan1000
        ipv4: 172.16.0.3/24
        node: s1
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
      vrf: red
    loopback:
      ipv4: 10.0.0.4/32
    mgmt:
      ifname: Management0
      ipv4: 192.168.121.104
      mac: 08-4F-A9-00-00-04
    module:
    - vlan
    - vrf
    name: s2
    vlan:
      max_bridge_group: 3
    vlans:
      blue:
        bridge_group: 1
        id: 1001
        mode: irb
        neighbors:
        - ifname: eth1.1001
          ipv4: 172.16.1.1/24
          node: r1
        - ifname: eth1.1001
          ipv4: 172.16.1.2/24
          node: r2
        - ifname: Vlan1001
          ipv4: 172.16.1.4/24
          node: s2
        - ifname: Vlan1001
          ipv4: 172.16.1.3/24
          node: s1
        - ifname: eth1
          ipv4: 172.16.1.6/24
          node: h2
        prefix:
          ipv4: 172.16.1.0/24
        vni: 101001
        vrf: blue
      green:
        bridge_group: 2
        id: 1002
        mode: route
        neighbors:
        - ifname: eth1.1002
          ipv4: true
          ipv6: true
          node: r1
        - ifname: eth1.1002
          ipv4: true
          ipv6: true
          node: r2
        - ifname: Vlan1002
          node: s2
        - ifname: Vlan1002
          node: s1
        - ifname: eth1
          ipv4: true
          ipv6: true
          node: h3
        pool: unnumbered
        prefix:
          ipv4: true
          ipv6: true
        vni: 101002
        vrf: green
      red:
        bridge_group: 3
        id: 1000
        mode: irb
        neighbors:
        - ifname: eth1.1000
          ipv4: 172.16.0.1/24
          node: r1
        - ifname: eth1.1000
          ipv4: 172.16.0.2/24
          node: r2
        - ifname: eth1
          ipv4: 172.16.0.5/24
          node: h1
        - ifname: Vlan1000
          ipv4: 172.16.0.3/24
          node: s1
        - ifname: Vlan1000
          node: s2
        prefix:
          ipv4: 172.16.0.0/24
        vni: 101000
        vrf: red
    vrf:
      as: 65000
    vrfs:
      blue:
        af:
          ipv4: true
        export:
        - '65000:2'
        id: 2
        import:
        - '65000:2'
        rd: '65000:2'
        vrfidx: 101
      green:
        export:
        - '65000:3'
        id: 3
        import:
        - '65000:3'
        rd: '65000:3'
        vrfidx: 102
      red:
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        rd: '65000:1'
        vrfidx: 100
provider: clab
vlans:
  blue:
    host_count: 0
    id: 1001
    neighbors:
    - ifname: eth1.1001
      ipv4: 172.16.1.1/24
      node: r1
    - ifname: eth1.1001
      ipv4: 172.16.1.2/24
      node: r2
    - ifname: Vlan1001
      ipv4: 172.16.1.4/24
      node: s2
    - ifname: Vlan1001
      ipv4: 172.16.1.3/24
      node: s1
    - ifname: eth1
      ipv4: 172.16.1.6/24
      node: h2
    prefix:
      ipv4: 172.16.1.0/24
    vni: 101001
    vrf: blue
  green:
    host_count: 0
    id: 1002
    mode: route
    neighbors:
    - ifname: eth1.1002
      ipv4: true
      ipv6: true
      node: r1
    - ifname: eth1.1002
      ipv4: true
      ipv6: true
      node: r2
    - ifname: Vlan1002
      node: s2
    - ifname: Vlan1002
      node: s1
    - ifname: eth1
      ipv4: true
      ipv6: true
      node: h3
    pool: unnumbered
    prefix:
      ipv4: true
      ipv6: true
    vni: 101002
    vrf: green
  red:
    host_count: 0
    id: 1000
    neighbors:
    - ifname: eth1.1000
      ipv4: 172.16.0.1/24
      node: r1
    - ifname: eth1.1000
      ipv4: 172.16.0.2/24
      node: r2
    - ifname: eth1
      ipv4: 172.16.0.5/24
      node: h1
    - ifname: Vlan1000
      ipv4: 172.16.0.3/24
      node: s1
    - ifname: Vlan1000
      node: s2
    prefix:
      ipv4: 172.16.0.0/24
    vni: 101000
    vrf: red
vrf:
  as: 65000
vrfs:
  blue:
    export:
    - '65000:2'
    id: 2
    import:
    - '65000:2'
    rd: '65000:2'
  green:
    export:
    - '65000:3'
    id: 3
    import:
    - '65000:3'
    rd: '65000:3'
  red:
    export:
    - '65000:1'
    id: 1
    import:
    - '65000:1'
    rd: '65000:1'
